src/zc/relation/searchindex.py

```diff
@@ -28,6 +28,15 @@
 _marker = object()
 
 
+class NotEqual(object):
+
+    def __lt__(self, other):
+        return False
+
+    def __gt__(self, other):
+        return False
```

Comment:
I'm uncomfortable when I see a class implementing a subset of the rich comparison methods.  What about `__le__`, `__ge__`?  I suppose the default identity-based `__eq__`/`__ne__` suffice here?

Reply:
Good point, `__gt__` is enough to make the tests pass, but that might depend on the object's position in the Btree and maybe other things as well.  c44686886d5a74031859e9a58019538b23075bd0  adds the missing methods.

Reply:
Uh I'm not smart enough to figure out the implications of an inconsistent set of comparison methods like this.

Reply:
Afaics the comparison is only ever used during insertion into the BTree, i.e. to find the right place to insert the new value. So it shouldn't matter that the object claims not to be smaller or greater than anything else. Using the empty string breaks things with Python 3.7+, however…